src/Strava/API/Service/REST.php

```diff
@@ -56,11 +66,14 @@ protected function getResult($response)
     {
         $status = $response->getStatusCode();
 
-        if ($status == 200 || $status == 201) {
-            return json_decode($response->getBody(), JSON_PRETTY_PRINT);
-        } else {
-            return [$status => $response->getReasonPhrase()];
-        }
+        $expandedResponse = [];
+
+        $expandedResponse['headers'] = $response->getHeaders();
+        $expandedResponse['body'] = json_decode($response->getBody(), JSON_PRETTY_PRINT);
+        $expandedResponse['success'] = $status == 200 || $status == 201;
```

Comment:
I think these comparison operators could be type-safe "===" instead of "==".

Reply:
I agree, now fixed in a new commit